generate.py

```diff
@@ -0,0 +1,172 @@
+        generated_text = self.convert_ids_to_text(generated_ids)
+        return generated_text
+
+    def generate_multi_track_sequence(
+        self,
+        inst_list=["INST=DRUMS", "INST=38", "INST=82"],
+        density_list=[3, 6, 2],
+    ):
+        generate_features_dict = {
+            "inst_list": inst_list,
+            "density_list": density_list,
+            "temperature": self.temperature,
+            "max_seq_length": self.max_length,
+            "generate_until": self.generate_until,
+        }
+
+        generated_multi_track_sequence = []
+        input = "PIECE_START"
+        for inst, density in zip(inst_list, density_list):
+            input = self.generate_one_sequence(
+                input=f"{input}", inst=inst, density=density
+            )
+        generated_multi_track_sequence = input
+        return generated_multi_track_sequence, generate_features_dict
+
+
+class WriteTextMidiToFile:  # utils saving to file
+    def __init__(self, sequence, output_path, feature_dict=None):
+        self.sequence = sequence
+        self.output_path = output_path
+        self.feature_dict = feature_dict
+
+    def hashing_seq(self):
+        self.filename = sha256(self.sequence.encode("utf-8")).hexdigest()
+        self.output_path_filename = f"{self.output_path}/{self.filename}.txt"
+
+    def writing_seq_to_file(self):
+        file_object = open(f"{self.output_path_filename}", "w")
+        assert type(self.sequence) is str, "sequence must be a string"
+        file_object.writelines(self.sequence)
+        file_object.close()
+        print(f"Token sequence written: {self.output_path_filename}")
+
+    def writing_feature_dict_to_file(self):
+        with open(f"{self.output_path}{self.filename}_features.json", "w") as json_file:
+            json.dump(self.feature_dict, json_file)
+
+    def text_midi_to_file(self):
+        self.hashing_seq()
+        self.writing_seq_to_file()
+        self.writing_feature_dict_to_file()
+
+
+if __name__ == "__main__":
+
+    device = "cpu"
+    # model path
+    model_path = "./models/model_2048_10kseq"
+    tokenizer_path = "./models/model_2048_10kseq/tokenizer.json"
+    generated_sequence_files_path = "models/model_2048_10kseq/generated_sequences/"
+
+    # load model and tokenizer
+    model = GPT2LMHeadModel.from_pretrained(model_path).to(device)
+    tokenizer = PreTrainedTokenizerFast(tokenizer_file=tokenizer_path)
+    temperature = 0.2
+
+    # instantiate the GenerateMidiText class
+    gen = GenerateMidiText(model, tokenizer, device, temperature=temperature)
+
+    # generate a multi track sequence
+    inst_list = ["INST=DRUMS", "INST=38", "INST=82", "INST=51"]
+    density_list = [3, 6, 2, 1]
+    (
+        generated_multi_track_sequence,
+        generate_features_dict,
+    ) = gen.generate_multi_track_sequence(
+        inst_list=inst_list,
+        density_list=density_list,
+    )
+
+    generated_sequence_files_path = f"{generated_sequence_files_path}"
```

Comment:
this also would be obsolete if we could use writeToFile function 👯‍♀️

Reply:
lol what is that line 🗡️